source/extensions/common/wasm/exports.cc

```diff
@@ -248,7 +248,7 @@ Word call_foreign_function(void* raw_context, Word function_name, Word function_
   if (!context->wasmVm()->setWord(results_size.u64_, Word(result_size))) {
     return wasmResultToWord(WasmResult::InvalidMemoryAccess);
   }
```

Comment:
If `res != WasmResult::Ok` then you probably shouldn't be writing the `result` above.

Reply:
@jplevyak could you fix that? Since this is already merged.

Reply:
Perhaps.  Again.  I will defer to reviewers.  I am OK providing more information to the caller assuming that the foreign function provided it as this that is part of a private API between the caller and the callee.

Reply:
It's fine to pass error details.

Reply:
Usually, when call fails (i.e. `res != WasmResult::Ok`), it means that results were not necessarily initialized, and `result` and `result_size` may point to garbage.